shell/platform/common/cpp/text_input_model.h

```diff
@@ -70,12 +70,12 @@ class TextInputModel {
   // Attempts to move the cursor to the beginning.
   //
   // Returns true if the cursor could be moved.
```

Comment:
I would say it's the comment that was wrong here, not the API. If it's already at the beginning, you've still succeeded in moving the cursor to the beginning. I would keep these void.

Reply:
I thought the intention of the return value is to know if an update needs sending, i.e.
```c++
if (model->MoveCursorToBeginning())
   sendUpdate();
```

`MoveCursorBack()` and `MoveCursorForward()` both behave like this.

Reply:
I'll commit this as-is, but please respond if you still think this should be different.